File: braggbnb108_ui/src/services/CleaningFeeService.js
import http from "../http-common"; 

class CleaningFeeService {
  getAllCleaningFees(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/cleaningFee/cleaningFees`, searchDTO);
  }

  get(cleaningFeeId) {
    return this.getRequest(`/cleaningFee/${cleaningFeeId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/cleaningFee?field=${matchData}`, null);
  }

  addCleaningFee(data) {
    return http.post("/cleaningFee/addCleaningFee", data);
  }

  update(data) {
  	return http.post("/cleaningFee/updateCleaningFee", data);
  }
  
  uploadImage(data,cleaningFeeId) {
  	return http.postForm("/cleaningFee/uploadImage/"+cleaningFeeId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new CleaningFeeService();

File: src/main/java/com/braggbnb108/service/CleaningFeeService.java
package com.braggbnb108.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.braggbnb108.domain.CleaningFee;
import com.braggbnb108.dto.CleaningFeeDTO;
import com.braggbnb108.dto.CleaningFeeSearchDTO;
import com.braggbnb108.dto.CleaningFeePageDTO;
import com.braggbnb108.dto.CleaningFeeConvertCriteriaDTO;
import com.braggbnb108.service.GenericService;
import com.braggbnb108.dto.common.RequestDTO;
import com.braggbnb108.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface CleaningFeeService extends GenericService<CleaningFee, Integer> {

	List<CleaningFee> findAll();

	ResultDTO addCleaningFee(CleaningFeeDTO cleaningFeeDTO, RequestDTO requestDTO);

	ResultDTO updateCleaningFee(CleaningFeeDTO cleaningFeeDTO, RequestDTO requestDTO);

    Page<CleaningFee> getAllCleaningFees(Pageable pageable);

    Page<CleaningFee> getAllCleaningFees(Specification<CleaningFee> spec, Pageable pageable);

	ResponseEntity<CleaningFeePageDTO> getCleaningFees(CleaningFeeSearchDTO cleaningFeeSearchDTO);
	
	List<CleaningFeeDTO> convertCleaningFeesToCleaningFeeDTOs(List<CleaningFee> cleaningFees, CleaningFeeConvertCriteriaDTO convertCriteria);

	CleaningFeeDTO getCleaningFeeDTOById(Integer cleaningFeeId);







}





